dna/pDna/pDna.go

```diff
@@ -0,0 +1,43 @@
+package pDna
+
+import "math"
+
+// Float32Base encodes a DNA base as a probability vector using float32 precision.
+// Note that gap probabilities can be stored implicitly as 1 - (A + C + G + T).
+type Float32Base struct {
+	A float32
+	C float32
+	G float32
+	T float32
+}
+
+// EqualBase returns true if two input Float32Base structs, p and q, are 'equal', defined
+// within a user-defined degree of precision.
+func EqualBase(p Float32Base, q Float32Base, precision float32) bool {
+	if !equalFloatPrecision(p.A, q.A, precision) {
+		return false
+	}
+	if !equalFloatPrecision(p.C, q.C, precision) {
+		return false
+	}
+	if !equalFloatPrecision(p.G, q.G, precision) {
+		return false
+	}
+	if !equalFloatPrecision(p.T, q.T, precision) {
+		return false
+	}
+	return true
+}
+
+// equalFloatPrecision is a helper function of EqualBase, which allows us to determine if
+// two float32 are 'equal' within a user-specified degree of precision. Note that this is a
+// relative precision (normalized to one of the values), and catches divide by zero errors.
+func equalFloatPrecision(a float32, b float32, precision float32) bool {
+	if a == 0 && b == 0 {
+		return true
+	}
+	if a == 0 {
```

Comment:
I think technically this should be else if a == 0, but since there is a return above it doesn't matter